src/components/subscription/SubscriptionItem.tsx

```diff
@@ -0,0 +1,82 @@
+import { SUBSCRIPTION_ITEM_STYLE } from '@/constants/styles';
+import type { Platform } from '@/types/recording';
+import { MinusIcon } from 'lucide-react';
+import { useState } from 'react';
+import ConfirmModal from '../common/ConfirmModal';
+import AlertModal from '../common/AlertModal';
+import useUnsubscribeMutation from '@/hooks/useUnsubscribeMutation';
+
+type SubscriptionItemProps = {
+  subscriptionId: number;
+  subscribedAt: string | null;
+  channel: {
+    channelId: string;
+    channelName: string;
+    platform: Platform;
+  };
+};
+
+const SubscriptionItem = (props: SubscriptionItemProps) => {
+  const { subscriptionId, subscribedAt, channel } = props;
+  const [alertMessage, setAlertMessage] = useState('');
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+
+  const confirmMessage = `[${channel.channelName}] 채널 구독을 해지하시겠습니까?`;
+
+  const { mutate: mutateUnsubscribe } = useUnsubscribeMutation(
+    () => setAlertMessage('채널 구독이 해지되었습니다.'),
+    () => setAlertMessage('구독채널 해지 도중 문제가 발생했습니다.')
+  );
```

Comment:
custom hook으로 빼서 관심사의 분리를 한 건 너무 좋은 것 같습니다!
다만, 처음 이 코드를 보는 경우, 콜백으로 어떤 걸 처리하는지 바로 보이지 않으니, 이렇게 해보는건 어떨까요??
```suggestion
  const { mutate: mutateUnsubscribe } = useUnsubscribeMutation(
    onSuccess: () => setAlertMessage('채널 구독이 해지되었습니다.'),
    onError: () => setAlertMessage('구독채널 해지 도중 문제가 발생했습니다.')
  );
```
가독성도 한번 생각해보면 좋을 것 같습니다!

Reply:
리뷰 감사합니다!
다시 보니 함수가 선언된 부분은 확인하지 않고는 어떤 역할을 하는 함수를 인자로 넘기는지 바로 확인이 어렵긴 하겠네요.
```suggestion
  const { mutate: mutateUnsubscribe } = useUnsubscribeMutation({
    onSuccess: () => setAlertMessage('채널 구독이 해지되었습니다.'),
    onError: () => setAlertMessage('구독채널 해지 도중 문제가 발생했습니다.'),
  });
```
이렇게 객체로 전달하는 방식 말쓸하시는 걸까요?

Reply:
네!! 어떤걸 콜백으로 넘기는지 알면 더 좋을 것 같습니다 :)